Tests/RegexTests/ParseTests.swift

```diff
@@ -2555,6 +2576,27 @@ extension RegexTests {
     diagnosticTest(#"\˂"#, .invalidEscape("˂"))
     diagnosticTest(#"\d\#u{301}"#, .invalidEscape("d\u{301}"))
 
+    // MARK: Confusable characters
+
+    diagnosticTest("[\u{301}]", .confusableCharacter("[\u{301}"))
+    diagnosticTest("(\u{358})", .confusableCharacter("(\u{358}"))
+    diagnosticTest("{\u{35B}}", .confusableCharacter("{\u{35B}"))
+    diagnosticTest(#"\\#u{35C}"#, .confusableCharacter(#"\\#u{35C}"#))
+    diagnosticTest("^\u{35D}", .confusableCharacter("^\u{35D}"))
+    diagnosticTest("$\u{35E}", .confusableCharacter("$\u{35E}"))
+    diagnosticTest(".\u{35F}", .confusableCharacter(".\u{35F}"))
+    diagnosticTest("|\u{360}", .confusableCharacter("|\u{360}"))
+    diagnosticTest(" \u{361}", .confusableCharacter(" \u{361}"))
```

Comment:
@milseman Should we include whitespace characters for this diagnostic? Not really metacharacters, but also you'd probably want to write out explicitly with `\u{...}` anyway.

Reply:
Combining characters with whitespace are pretty nasty and satisfy (one of) Unicode's definition of degenerate. So I'm good forcing them to use explicit scalar escapes too.